hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/types/DatanodePipeline.java

```diff
@@ -17,22 +17,22 @@
  */
 package org.apache.hadoop.ozone.recon.api.types;
 
-import org.apache.hadoop.hdds.client.ReplicationConfig;
-
 import java.util.UUID;
 
 /**
  * Metadata object that contains pipeline information of a Datanode.
  */
 public class DatanodePipeline {
   private UUID pipelineID;
-  private ReplicationConfig repConfig;
+  private String replicationType;
+  private String replicationFactor;
   private String leaderNode;
 
-  public DatanodePipeline(UUID pipelineID, ReplicationConfig repConfig,
-      String leaderNode) {
+  public DatanodePipeline(UUID pipelineID, String replicationType,
```

Comment:
If we cannot store the ReplicationConfig object in the class, perhaps we could pass ReplicationConfig to the constructor, and inside the constructor pull out the String values and store them? In general, we are trying to get away from Factor and Type in the API everywhere, so that would be a good compromise perhaps?

Reply:
I updated the patch that way. Let me know if this is ok.